bundle/edu.gemini.ags/src/main/scala/edu/gemini/ags/api/AgsStrategy.scala

```diff
@@ -85,12 +85,30 @@ object AgsStrategy {
      * Creates a new TargetEnvironment with guide stars for each assignment in
      * the Selection.
      */
-    def applyTo(env: TargetEnvironment): TargetEnvironment =
+    def applyTo(env: TargetEnvironment): TargetEnvironment = {
+      def findMatching(gpt: GuideProbeTargets, target: SPTarget): Option[SPTarget] =
+        Option(target.getTarget.getName).flatMap { n =>
+          gpt.getTargets.asScalaList.find { t =>
+            Option(t.getTarget.getName).map(_.trim).exists(_ == n.trim)
```

Comment:
Or we might just pull out the name fetching and trimming since it is done twice.  Is name matching sufficient?  Seems like it would be but I don't remember how it used to work.

````scala
  def name(t: SPTarget): Option[String] = Option(t.getTarget.getName).map(_.trim)
````